### ISSUE TITLE
Generated images are almost entirely black.

### ISSUE THREAD

**Issue by 1359347500cwc** [LTH14/mar#80]
I am using my self-trained autoencoder as both the encoder and decoder to train on the ImageNet dataset. After 270 epochs, the loss has decreased to around 0.3, but the generated images are almost entirely black.

here is the eval.sh

torchrun
--node_rank=$RANK
--nproc_per_node=8
--nnodes=1
--master_addr="${MASTER_ADDR}"
--master_port=12344
main_mar.py
--model mar_base --diffloss_d 6 --diffloss_w 1024
--eval_bsz 32 --num_images 50000
--num_iter 256 --num_sampling_steps 100 --cfg 2.9 --cfg_schedule linear --temperature 1.0
--output_dir /mar_result/mar_mamba_base_0208
--resume base_0208/checkpoint-270.pth
--data_path ${IMAGENET_PATH} --evaluate

**Comment by 1359347500cwc** [comment]
I print the sampe_tokens's values

all the values are nan

`                sampled_tokens = model_without_ddp.sample_tokens(bsz=batch_size, num_iter=args.num_iter, cfg=cfg,
                                                                 cfg_schedule=args.cfg_schedule, labels=labels_gen,
                                                                 temperature=args.temperature)
                
                # 计算统计特征
                print("Shape:", sampled_tokens.shape)
             

                # Convert to numpy for easier statistics if needed
                sampled_tokens_np = sampled_tokens.detach().cpu().numpy()  # Detach from the graph and move to CPU

                mean = sampled_tokens_np.mean()
                std_dev = sampled_tokens_np.std()
                min_value = sampled_tokens_np.min()
                max_value = sampled_tokens_np.max()
`
[07:55:19.963498] Shape: torch.Size([32, 16, 16, 16])
[07:55:19.969298] Mean: nan
[07:55:19.969347] Standard Deviation: nan
[07:55:19.969365] Min Value: nan
[07:55:19.969384] Max Value: nan

**Comment by LIRENDA621** [comment]
Hi, I have the same issue, how do you fix this? " loss has decreased to around 0.3, but the generated images are almost entirely black"

**Comment by LTH14** [comment]
We use ema=0.9999 to generate image, so you need to wait for 160 epochs on ImageNet (around 100K training steps) to see reasonable results.

**Comment by 1359347500cwc** [comment]
> Hi, I have the same issue, how do you fix this? " loss has decreased to around 0.3, but the generated images are almost entirely black"

In my case, I specified the wrong Resume path. The Resume path should specify the saved Dir instead of a specific checkpoint path.  **“--resume base_0208/checkpoint-270.pth”** is Wrong  **“--resume base_0208/”** is right
